src/__rt/future.rs

```diff
@@ -0,0 +1,51 @@
+use std::mem;
+use std::ops::{Generator, GeneratorState};
+
+use super::{IsResult, Mu, Reset, CTX};
+
+use futures::task;
```

Comment:
Use `futures-core` here and elsewhere rather than `futures`-- this code doesn't rely on the combinators, and it'd be nice to package `futures-await` into the `futures` crate itself behind a `nightly` feature or something.

Reply:
Currently `futures-await` is designed to be a replacement, rather than extension, for futures. Its designed to be used as:

```rust
extern crate futures_await as futures;
```

So it re-exports all of futures.

Changes to this structure I think are separate from this PR

Reply:
Okay. We can try and catch it in follow-ups, then.